src/main/java/net/neoforged/forge/NeoForgeMod.java

```diff
@@ -0,0 +1,9 @@
+package net.neoforged.forge;
+
+import net.minecraftforge.fml.common.Mod;
+
+@Mod("neoforge")
+public class NeoForgeMod
+{
+    // TODO: Move relevant stuff from ForgeMod here once backward compatibility is no longer needed
```

Comment:
Does this bring any value to modders? Seems like a pointless breaking change to me.

Reply:
It is something to be considered and discussed.
As soon as NeoForge decides to stop being compatible, there's a strong argument (IMO) to be made in favor of doing a blanket package rename away from net.minecraftforge, so that changes in either loader cannot create sudden unexpected incompatibilities.
Having the main mod file be a new file felt like the right thing to do to me, but I can be convinced otherwise. However, we do need somewhere to add the "forge" mod annotation if the "neoforge" modId gets assigned to the ForgeMod class, so a new class would be needed either way.
Although that said, I have been told there's plans for an aliasing system which would make this change completely redundant, so I will await more information about that and possibly change/revert this, or close this PR in favor of another one that leverages that feature. Meanwhile, I want to figure out what exactly is needed in order to get this working, so that the version/modid change can be soon as soon as possible.